src/zenml/model/utils.py

```diff
@@ -0,0 +1,150 @@
+#  permissions and limitations under the License.
+"""Utility functions for linking step outputs to model versions."""
+
+from typing import (
+    Dict,
+)
+from uuid import UUID
+
+from zenml.artifacts.artifact_config import ArtifactConfig
+from zenml.client import Client
+from zenml.exceptions import StepContextError
+from zenml.logger import get_logger
+from zenml.model.model_config import ModelConfig
+from zenml.models.model_models import (
+    ModelVersionArtifactFilterModel,
+    ModelVersionArtifactRequestModel,
+)
+from zenml.new.steps.step_context import get_step_context
+
+logger = get_logger(__name__)
+
+
+def link_step_artifacts_to_model(
+    artifact_ids: Dict[str, UUID],
+) -> None:
+    """Links the output artifacts of a step to the model.
+
+    Args:
+        artifact_ids: The IDs of the published output artifacts.
+
+    Raises:
+        RuntimeError: If called outside of a step.
+    """
+    try:
+        step_context = get_step_context()
+    except StepContextError:
+        raise RuntimeError(
+            "`link_artifacts_to_model` can only be called from within a step."
+        )
+    try:
+        model_config = step_context.model_config
+    except StepContextError:
+        model_config = None
+        logger.debug("No model context found, unable to auto-link artifacts.")
+
+    for artifact_name in artifact_ids:
+        artifact_id = artifact_ids[artifact_name]
+        artifact_config = step_context._get_output(
+            artifact_name
+        ).artifact_config
+        if artifact_config is None and model_config is not None:
+            artifact_config = ArtifactConfig(
+                name=artifact_name,
+                model_name=model_config.name,
+                model_version=model_config.version,
+            )
+            logger.info(
+                f"Implicitly linking artifact `{artifact_name}` to model "
+                f"`{model_config.name}` version `{model_config.version}`."
+            )
+
+        if artifact_config and model_config:
+            link_artifact_config_to_model(
+                artifact_config=artifact_config,
+                artifact_name=artifact_name,
+                artifact_id=artifact_id,
+                model_config=model_config,
+            )
+
+
+def link_artifact_config_to_model(
+    artifact_config: ArtifactConfig,
+    model_config: "ModelConfig",
+    artifact_name: str,
+    artifact_id: UUID,
+) -> None:
+    """Link an artifact config to a model version.
+
+    Args:
+        artifact_config: The artifact config to link.
+        model_config: The model config to link the artifact to.
+        artifact_name: The name of the artifact to link.
+        artifact_id: The ID of the artifact to link.
+    """
+    client = Client()
+
+    artifact_name = artifact_config.name or artifact_name
+    if artifact_name is None:
+        artifact = client.zen_store.get_artifact(artifact_id=artifact_id)
+        artifact_name = artifact.name
```

Comment:
For my understanding: when `or artifact_name` come into play? Why If I don't have it in the config, I'd assume to get it from the artifact as a standard name, or?

Reply:
For backwards compatibility to still allow the old syntax of specifying name outside of artifact config like this: `Annotated[int, "name", ArtifactConfig(...)]`

Reply:
Yeah I think we don't need it actually